atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -273,6 +290,9 @@ public <T, E extends Exception> T finishWithCallback(TransactionTask<T, E> task,
                 postTaskContext = postTaskTimer.time();
                 timelockService.tryUnlock(ImmutableSet.of(immutableTsLock));
                 openTransactionCounter.dec();
+                // need a fix for if the user does not call finishWithCallback
```

Comment:
It's part of the contract that users must do this. You can assume they are using the API correctly